nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2121,21 +2351,29 @@ mod test {
         };
         println!("1 -> 2: marked non-provisionable {nonprovisionable_sled_id}");
         let expunged_sled_id = {
-            let (sled_id, details) = sleds_iter.next().expect("no sleds");
-            details.policy = SledPolicy::Expunged;
+            let (sled_id, _) = sleds_iter.next().expect("no sleds");
+            // We need to call builde.expunge, but can't while iterating; we defer
```

Comment:
```suggestion
            // We need to call builder.expunge_sled(), but can't while iterating; we defer
```

Reply:
Fixed in de4a0805309651eea73e6c8bab1e256afbaf992d